Algorithm/QCAlgorithm.cs

```diff
@@ -526,6 +526,11 @@ public void PostInitialize()
                     // if it wasn't manually added, add a subscription for underlying updates
                     equity = AddEquity(underlying.Value, option.Resolution, underlying.ID.Market, false);
                 }
+                // In the options trading, the strike price, the options settlement and exercise are
+                // all based on the raw price of the underlying asset instead of the adjusted price. 
+                // In order to select the accurate contracts, we need to set
+                // the data normalization mode of the underlying asset to be raw
+                equity.SetDataNormalizationMode(DataNormalizationMode.Raw);
```

Comment:
If the equity isn't already raw mode -> please send a Debug/Warning message notifying the user of this behaviour. 
"The AAPL equity security was set the raw price normalization mode to work with options."